.github/workflows/github-actions-demo.yaml

```diff
@@ -0,0 +1,24 @@
+name: GitHub Actions Demo
+on:
+  push:
+    branches: [ $default-branch ]
```

Comment:
This variable `$default-branch` is not supported in Workflows, but only in Workflow Templates.

Ref: 
https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/
https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization

Reply:
Correct. Some more reference: https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable